development/applications/hedwig/src/flight_control.c

```diff
@@ -222,17 +226,23 @@ void flight_control_get_outputs(fpr_status_t *status_response)
   }
 }
 
-void flight_control_set_mode(fe_flight_mode_t new_mode)
+int flight_control_set_mode(fe_flight_mode_t new_mode)
 {
   switch (new_mode) {
     case FE_FLIGHT_MODE_IDLE:
-      if (flight_control_mode == FE_FLIGHT_MODE_FCS_READY) {
+      if (flight_control_mode != FE_FLIGHT_MODE_CALIBRATING) {
+        if (lock_output_data() == pdTRUE) {
+          motors_off();
+          unlock_output_data();
```

Comment:
this can be cleaned up a lot by letting the cases that are the same fall through:

```
int flight_control_set_mode(fe_flight_mode_t new_mode)
{
  switch (new_mode) {
    ...
    case FE_FLIGHT_MODE_CALIBRATING:
    case FE_FLIGHT_MODE_FLY:
      if (flight_control_mode == FE_FLIGHT_MODE_FCS_READY) {
        flight_control_mode = new_mode;
        return FLN_OK;
      }
      break;
  }
  return FLN_ERR;
}


```